src/routes/api/playlists/user/+server.ts

```diff
@@ -1,14 +1,29 @@
-import { authenticate, readPath } from '$lib/server/firebaseUtils';
+import { authenticate, getOrganizationInfo, getOrganizationPlaylists, getUserOrganizations, modifyPlaylistsResponse, readPath } from '$lib/server/firebaseUtils';
 import { json } from '@sveltejs/kit';
 import type { RequestHandler } from './$types';
 
 export const GET = (async (event) => {
   const { uid } = await authenticate(event);
-  const playlists = (await readPath(`/playlists/user/${uid}`)) || {};
-  const modifiedPlaylists = Object.values(playlists).map((playlist) => ({
-    ...playlist,
-    words: playlist.words.map((word) => Object.values(word).map((letters) => (letters === false ? null : letters))),
-  }));
+  const userPlaylists = (await readPath<Database.Playlists.User>(`/playlists/user/${uid}`)) || {};
+  const userPlaylistsArray = Object.values(userPlaylists).map((playlist) => playlist);
+  const organizationIds = await getUserOrganizations(uid);
+  const organizationPlaylistsArray = (
+    await Promise.all(
+      organizationIds.map(async (orgId) => {
+        const orgPlaylists = (await getOrganizationPlaylists(orgId) ?? {});
+        const orgInfo = await getOrganizationInfo(orgId);
+        return Object.values(orgPlaylists).map((playlist) => ({
+          ...playlist.playlist,
+          orgSource: {
+            orgName: orgInfo?.name,
+            orgId,
+          },
+        }))
+      })
+    )
+  ).flat();
```

Comment:
If this is slow at all we could parallelize it into two workstreams, user and org playlists. Don't need to prematurely optimize though, we'll see how it runs.